app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/MainActivity.kt

```diff
@@ -1,22 +1,40 @@
 package com.github.HumanLearning2021.HumanLearningApp
 
+import android.app.Dialog
+import android.content.DialogInterface
 import android.content.Intent
-import androidx.appcompat.app.AppCompatActivity
-
+import android.graphics.drawable.Drawable
 import android.os.Bundle
 import android.view.View
+import android.widget.ImageView
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
 import com.github.HumanLearning2021.HumanLearningApp.view.DisplayDatasetActivity
 import com.github.HumanLearning2021.HumanLearningApp.view.LearningActivity
 
+
 class MainActivity : AppCompatActivity() {
+    val addPicture = registerForActivityResult(AddPictureActivity.AddPictureContract) { result ->
+        if (result != null) {
+            val img = ImageView(this)
+            img.setImageDrawable(Drawable.createFromPath(result.second.path))
+            val builder: AlertDialog.Builder =
+                AlertDialog.Builder(this).apply {
+                    setMessage(result.first)
+                    setPositiveButton("OK"
+                    ) { dialog, which -> dialog.dismiss() }.setView(img)
+                }
+            builder.create().show()
+        }
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_main)
     }
 
     fun launchToCameraActivity(view: View) {
-        val intent = Intent(this, AddPictureActivity::class.java)
-        startActivity(intent)
+        addPicture.launch(arrayOf("selfie", "yellow head", "some guy"))
```

Comment:
If we put something like door, sofa, and cat, it matches the virtual room the virtual devices can see. It would be just that bit cooler for the demo wouldn't it?

Reply:
That's an awesome idea ! and easy to modify.

Reply:
On my device it is a face

Reply:
oh that's odd. If possible taylor it to whoever does the demo

Reply:
Wait maybe you're using the front camera?

Reply:
![Capture d’écran de 2021-03-18 23-53-48](https://user-images.githubusercontent.com/15657735/111707948-449a0e80-8845-11eb-8e89-7adb944f9738.png)
This is the selfie camera on my Pixel 3 API 29. Is that the same thing you see?

To change camera we have to use the camera app right now I think.

Reply:
Yeah, I just came to the exact same conclusion